ivy/functional/backends/paddle/elementwise.py

```diff
@@ -494,15 +342,11 @@ def divide(
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
     x1, x2, ret_dtype = _elementwise_helper(x1, x2)
-    if x1.dtype in [paddle.float16, paddle.bfloat16]:
-        x1, x2 = x1.astype("float32"), x2.astype("float32")
-    if not (ivy.is_float_dtype(ret_dtype) or ivy.is_complex_dtype(ret_dtype)):
-        ret_dtype = ivy.default_float_dtype(as_native=True)
```

Comment:
can we get `complex` dtypes to work in the same way as other functions for this function too (i.e. using `is_complex` and then calling the function for `real` and `complex`)?

Reply:
sure